packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,98 +9,319 @@ import 'package:file/file.dart';
 
     // Calculate the appropriate basedir for the current test context.
     final FileSystem fs = goldens.fs;
     final Directory testDirectory = fs.directory(defaultComparator.basedir);
     final String testDirectoryRelativePath = fs.path.relative(testDirectory.path, from: goldens.flutterRoot.path);
-    return FlutterGoldenFileComparator(goldens.repositoryRoot.childDirectory(testDirectoryRelativePath).uri);
+    return FlutterGoldensRepositoryFileComparator(goldens.comparisonRoot.childDirectory(testDirectoryRelativePath).uri);
   }
 
+  /// Compares [imageBytes] against the golden file identified by [golden].
+  ///
+  /// The returned future completes with a boolean value that indicates whether
+  /// [imageBytes] matches the golden file's bytes.
+  ///
+  /// In the case of the given golden file being unavailable, the comparator
+  /// will throw a [TestFailure].
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
-    final File goldenFile = _getGoldenFile(golden);
+    final File goldenFile = getGoldenFile(golden);
     if (!goldenFile.existsSync()) {
       throw TestFailure('Could not be compared against non-existent file: "$golden"');
     }
-    final List<int> goldenBytes = await goldenFile.readAsBytes();
-    // TODO(tvolkert): Improve the intelligence of this comparison.
-    if (goldenBytes.length != imageBytes.length) {
-      return false;
-    }
-    for (int i = 0; i < goldenBytes.length; i++) {
-      if (goldenBytes[i] != imageBytes[i]) {
+
+    if (platform.isLinux) {
+      final List<int> goldenBytes = await goldenFile.readAsBytes();
+      if (goldenBytes.length != imageBytes.length) {
         return false;
       }
+      for (int i = 0; i < goldenBytes.length; i++) {
+        if (goldenBytes[i] != imageBytes[i]) {
+          return false;
+        }
+      }
+      return true;
     }
+    print('Skipping "$golden" : Skia Gold unavailable && !isLinux');
     return true;
   }
 
+  /// Updates the golden file identified by [golden] with [imageBytes].
+  ///
+  /// This will be invoked in lieu of [compare] when [autoUpdateGoldenFiles]
+  /// is `true` (which gets set automatically by the test framework when the
+  /// user runs `flutter test --update-goldens`).
+  @override
+  Future<void> update(Uri golden, Uint8List imageBytes) async {
+    final File goldenFile = getGoldenFile(golden);
+    await goldenFile.parent.create(recursive: true);
+    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  }
+
+  /// Returns the golden file identified by the given [Uri].
+  @override
+  File getGoldenFile(Uri uri) {
+    return fs.directory(basedir).childFile(fs.file(uri).path);
+  }
+
+  /// Modifies the given [Uri] to incorporate an optional [version] number.
+  ///
+  /// Golden files for package:flutter use version numbers, and must be used to
+  /// reference the appropriate file.
+  @override
+  Uri getTestUri(Uri key, int version) {
+    return _addVersion(key, version);
```

Comment:
Just inline `_addVersion`? It is not that complicated and not really used anywhere else?